http/encoding.go

```diff
@@ -97,22 +103,58 @@ func ResponseEncoder(ctx context.Context, w http.ResponseWriter) Encoder {
 		}
 		return nil, ""
 	}
+	negotiateCT := func(ct string) Encoder {
```

Comment:
is this really needed? negotiateCT is only called once it seems. Also "negotiate" is not really a good term here as there is no negotiation happening between the client and the server.